### ISSUE TITLE
Expose a few WCS utils functions on SkyMap

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#632]
There's a few WCS utils standalone functions in Astropy:
http://docs.astropy.org/en/latest/wcs/index.html#id4

These aren't methods on the WCS object, because WCS is very old and implemented partly in C, and SkyCoord, Quantity were only added a few years ago.

I would like to suggest we expose some of these convenience WCS utils functions on the `SkyMap` class (always prefixed with `wcs_` to have them grouped and easy to find via tab completion), to make it more convenient to use them (and for users to find them).

Concretely I'd suggest we start by adding these:
- [proj_plane_pixel_scales](http://docs.astropy.org/en/latest/api/astropy.wcs.utils.proj_plane_pixel_scales.html) -> `SkyMap.wcs_proj_plane_pixel_scales`
- [skycoord_to_pixel](http://docs.astropy.org/en/latest/api/astropy.wcs.utils.skycoord_to_pixel.html) -> `SkyMap.wcs_skycoord_to_pixel`
- [pixel_to_skycoord](http://docs.astropy.org/en/latest/api/astropy.wcs.utils.pixel_to_skycoord.html) -> `SkyMap.wcs_pixel_to_skycoord`

There might be others in the future, but not many more.

@adonath @OlgaVorokh - Thoughts?

**Comment by adonath** [comment]
At first glance I don't see a real advantage in this. It's just a slight change to the API and I can't see the one solution being more convenient over the other, I think it doesn't make a difference:
```
scale = proj_plane_pixel_scales(SkyMap.wcs)
# vs.
scale = SkyMap.wcs_proj_plane_pixel_scales()
```
Or:
```
pixel = skycoord_to_pixel(SkyMap.wcs, skycoord)
# vs.
pixel = SkyMap.wcs_skycoord_to_pixel(skycoord)
```
I agree it might be easier to find those methods, but I would rather encourage people to use astropy,
when the equivalent functionality can be found there, instead of re-exposing the functionality in `gammapy` with a minimal change in API.

I think ideally this would be additional methods on the `WCS` objects.

**Comment by cdeil** [comment]
You save the import. Astropy has a bad API here for historical reasons. Why not make this better for Gammapy users?

**Comment by adonath** [comment]
As I said in my previous comment, I don't think it makes a real difference in API or convenience to the user, that would justify spending the time implementing, testing, documenting and maintaining the code in gammapy (and even discussing;-). So i would rather say "If it ain't broke don't fix it" and spend the time with something else... if you feel strongly about it @cdeil or @OlgaVorokh, I don't have any objections if you want to work on that.

**Comment by cdeil** [comment]
> I agree it might be easier to find those methods, but I would rather encourage people to use astropy,
when the equivalent functionality can be found there, instead of re-exposing the functionality in gammapy with a minimal change in API.

To a large degree, `SkyMap` is a convenience wrapper around Numpy and Astropy, combining the numpy.array data and WCS object, re-exposing existing functionality with a better, more convenient API now that e.g. SkyCoord (and soon [PixCoord](http://astropy-regions.readthedocs.io/en/latest/api/regions.PixCoord.html
) is available.

> if you feel strongly about it @cdeil or @OlgaVorokh, I don't have any objections if you want to work on that.

Well, yes, I really want it.

I think now that SkyCoord and PixCoord are available, we should use that.
It's not easy and in some places not possible to retrofit WCS in a backward-compat manner, this is not going to happen. Wrapping in a new better API is a standard design pattern to handle this.

@OlgaVorokh or I will add it.

**Comment by adonath** [comment]
Resolved by #644 and #643
